commit 89fc6dab71b106e7d9f81bd29802ec20ee01bd5f
Author: Fabio M. Di Nitto <fdinitto@redhat.com>
Date:   Mon Aug 29 11:09:41 2011 +0200

    config: fix handling of temporary directory in ccs_update_schema
    
    tempdir was not set/exported and deleted correctly, leaving clutter
    in /tmp
    
    Resolves: rhbz#680930
    
    Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

diff --git a/config/tools/xml/ccs_update_schema.in b/config/tools/xml/ccs_update_schema.in
index e149896..03dbb68 100644
--- a/config/tools/xml/ccs_update_schema.in
+++ b/config/tools/xml/ccs_update_schema.in
@@ -56,9 +56,15 @@ if [ "$?" != 0 ]; then
 fi
 check_opts $opts
 
+tmpdir="$(mktemp -d -q)" || {
+	echo "unable to create tempdir" >&2
+	exit 1
+}
+export tmpdir
+
 # need to be careful (might have to mask traps on exit)
 cleanup() {
-	[ -n "$tmpdir" ] && [ -d "$tmddir" ] && rm -rf "$tmpdir"
+	[ -n "$tmpdir" ] && [ -d "$tmpdir" ] && rm -rf "$tmpdir"
 	rm -f $vardir/rng_update.lock
 	exit $1
 }
@@ -342,11 +348,6 @@ install_schema() {
 (
 	flock --exclusive 200
 
-	tmpdir="$(mktemp -d -q)" || {
-		echo "unable to create tempdir" >&2
-		exit 1
-	}
-
 	lecho "Generating resource-agents cache"
 
 	generate_ras "$tmpdir" || {
